vue-components/tests/unit/components/ExtendedNumberInput.spec.ts

```diff
@@ -1,36 +1,119 @@
 import { mount } from '@vue/test-utils';
 import ExtendedNumberInput from '@/components/ExtendedNumberInput.vue';
 import ValidationMessage from '@/components/ValidationMessage.vue';
+import Vue from 'vue';
 
 describe( 'ExtendedNumberInput', () => {
 
-	it( 'takes an input', () => {
-		const content = '123';
+	it( 'reconstructs its content from the value prop', async () => {
+		const content = { number: 123, precision: 0.5 };
 		const wrapper = mount( ExtendedNumberInput, {
 			propsData: {
 				value: content,
 			},
 		} );
+		await Vue.nextTick();
+
+		const inputElement = wrapper.find( 'input' );
+
+		expect( ( inputElement.element as HTMLFormElement ).value ).toBe( '123+-0.5' );
+	} );
+
+	it( 'mount with value null means empty string as initial input', async () => {
+		const wrapper = mount( ExtendedNumberInput, {
+			propsData: {
+				value: null,
+			},
+		} );
+		await Vue.nextTick();
+
 		const inputElement = wrapper.find( 'input' );
 
-		expect( ( inputElement.element as HTMLFormElement ).value ).toBe( content );
+		expect( ( inputElement.element as HTMLFormElement ).value ).toBe( '' );
+	} );
+
+	it( 'set different value after mount', async () => {
+		const wrapper = mount( ExtendedNumberInput );
+
+		const inputElement = wrapper.find( 'input' );
+		expect( ( inputElement.element as HTMLFormElement ).value ).toBe( '' );
+
+		wrapper.setProps( { value: { number: 123, precision: 0.5 } } );
+		await Vue.nextTick();
```

Comment:
taking from Amir's suggestion [here](https://github.com/wmde/query-builder/pull/214) , 
we can get rid of these `Vue.nexttrick()` things. shouldn't be too much effort

Reply:
Good point, fixed